src/new-client/src/App/LiveRates/Tiles/AnalyticsTile.tsx

```diff
@@ -1,13 +1,32 @@
-import { useState } from "react"
 import styled from "styled-components/macro"
-import {
-  LineChart,
-  ResponsiveContainer,
-  Line,
-  YAxis,
-  ReferenceLine,
-} from "recharts"
-import { useHistoricalPrices, getHistoricalPrices$ } from "services/prices"
+import { getHistoricalPrices$ } from "services/prices"
+import { curveCatmullRom, line, scaleLinear, scaleTime } from "d3"
+import { map } from "rxjs/operators"
+import { bind } from "@react-rxjs/core"
+
+const curvedLine = line<[Date, number]>().curve(curveCatmullRom)
+
+const VIEW_BOX_WIDTH = 200
+const VIEW_BOX_HEIGHT = 81
+
+export const xScale = scaleTime().range([0, VIEW_BOX_WIDTH])
+export const yScale = scaleLinear().range([0, VIEW_BOX_HEIGHT])
+
+const [useHistoricalPath, analyticsTile$] = bind((symbol: string) =>
+  getHistoricalPrices$(symbol).pipe(
+    map((prices) => {
+      const points = prices.map(
```

Comment:
Merit an abstraction?

Reply:
I was on the fence on this one... I've added a new commit that abstracts the main transformations of the "historical chart", in a way that they can be easily composed. I think that's worth it, let me know what you think about it. Thanks!